jerry-core/ecma/builtin-objects/ecma-builtin-function.cpp

```diff
@@ -56,6 +56,85 @@ ecma_builtin_function_dispatch_call (const ecma_value_t *arguments_list_p, /**<
 } /* ecma_builtin_function_dispatch_call */
 
 /**
+ * Helper method to count and convert the arguments for the Function constructor call.
+ *
+ * Performs the operation described in ECMA 262 v5.1 15.3.2.1 steps 5.a-d
+ *
+ *
+ * @return completion value - concatenated arguments as a string.
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+static ecma_completion_value_t
+ecma_builtin_function_helper_get_arguments (const ecma_value_t *arguments_list_p,
+                                            ecma_length_t arguments_list_len,
+                                            ecma_length_t *total_number_of_function_args)
```

Comment:
@galpeter, by `out_` prefix I mean `out_total_number_of_function_args_p`, may be `out_total_number_of_args_p`.